assets/src/dashboard/app/views/exploreTemplates/header/index.js

```diff
@@ -105,14 +104,42 @@ function Header({ filter, totalTemplates, search, templates, sort, view }) {
             onChange={onNewColorSelected}
           />
         </HeadingDropdownsContainer>
+      );
+    }
+    return null;
+  }, [
+    enableInProgressTemplateActions,
+    selectedCategories,
+    clearAllCategories,
+    onNewCategorySelected,
+    selectedColors,
+    clearAllColors,
+    onNewColorSelected,
+  ]);
+
+  return (
+    <Layout.Squishable>
+      <PageHeading
+        centerContent
+        defaultTitle={__('Templates', 'web-stories')}
+        searchPlaceholder={__('Search Templates', 'web-stories')}
+        stories={templates}
+        handleTypeaheadChange={
+          enableInProgressTemplateActions ? search.setKeyword : undefined
+        }
+        typeaheadValue={search.keyword}
+      >
+        {TemplateFilters}
       </PageHeading>
       <BodyViewOptions
         resultsLabel={resultsLabel}
         layoutStyle={view.style}
         handleLayoutSelect={view.toggleStyle}
         currentSort={sort.value}
         pageSortOptions={TEMPLATES_GALLERY_SORT_MENU_ITEMS}
-        handleSortChange={sort.set}
+        handleSortChange={
+          enableInProgressTemplateActions ? sort.set : undefined
```

Comment:
Could we change `undefined` here and on line `128` to be a noop function just for extra safety?